### ISSUE TITLE
Bazel tests failed to build on JDK7

### ISSUE THREAD

**Issue by meteorcloudy** [bazelbuild/bazel#1771]
http://ci.bazel.io/view/Bazel%20bootstrap%20and%20maintenance/job/Bazel/JAVA_VERSION=1.7,PLATFORM_NAME=windows-x86_64/815/consoleFull
```
ERROR: C:/jenkins/workspace/Bazel/JAVA_VERSION/1.7/PLATFORM_NAME/windows-x86_64/src/test/java/com/google/devtools/build/lib/BUILD:531:1: Couldn't build file src/test/java/com/google/devtools/build/lib/libpackages_testutil.jar: Java compilation in rule '//src/test/java/com/google/devtools/build/lib:packages_testutil' failed: java.exe failed: error executing command external/local_jdk/bin/java.exe -Xbootclasspath/p:external/bazel_tools/third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar ... (remaining 2 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
src\test\java\com\google\devtools\build\lib\packages\util\PackageLoadingTestCase.java:132: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        UUID.randomUUID(), ImmutableMap.of(),
                                          ^
src\test\java\com\google\devtools\build\lib\packages\util\PackageLoadingTestCase.java:146: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        ImmutableMap.of(),
                       ^
Note: Some messages have been simplified; recompile with -Xdiags:verbose to get full output
2 errors
BazelJavaBuilder threw exception: java compilation returned status ERROR
WARNING: Worker strategy cannot execute this Javac action, because the action's execution info does not contain 'supports-workers=1'.
____From Building src/test/java/com/google/devtools/build/lib/syntax_test.jar (24 source files):
```
culprit: 6f33a1c54e517d7343c36d0479713655a19f3224

**Comment by meteorcloudy** [comment]
Same errors in different classes, please fix :)
```
ERROR: C:/jenkins/workspace/Bazel/JAVA_VERSION/1.7/PLATFORM_NAME/windows-x86_64/src/test/java/com/google/devtools/build/lib/BUILD:298:1: Couldn't build file src/test/java/com/google/devtools/build/lib/libanalysis_testutil.jar: Java compilation in rule '//src/test/java/com/google/devtools/build/lib:analysis_testutil' failed: java.exe failed: error executing command external/local_jdk/bin/java.exe -Xbootclasspath/p:external/bazel_tools/third_party/java/jdk/langtools/javac.jar -XX:+TieredCompilation -XX:TieredStopAtLevel=1 -jar ... (remaining 2 argument(s) skipped): com.google.devtools.build.lib.shell.BadExitStatusException: Process exited with status 1.
src\test\java\com\google\devtools\build\lib\analysis\util\AnalysisTestCase.java:186: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        ImmutableMap.of(),
                       ^
src\test\java\com\google\devtools\build\lib\analysis\util\AnalysisTestCase.java:290: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        ImmutableMap.of(),
                       ^
src\test\java\com\google\devtools\build\lib\analysis\util\BuildViewTestCase.java:235: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        ImmutableMap.of(),
                       ^
src\test\java\com\google\devtools\build\lib\analysis\util\ConfigurationTestCase.java:126: error: incompatible types: ImmutableMap<Object,Object> cannot be converted to Map<String,String>
        ImmutableMap.of(),
```

**Comment by aehlig** [comment]
Fixed by 03b9cfd3f9117fc22bf8ae129c316cf606b46a00 and 15e5cdc3e4e1c931673427e1be216eee633eda57
